### PR TITLE
feat: Add defaultDiskSelector option to chart

### PR DESCRIPTION
**Pull request by rafaelgaspar**
#### What this PR does / why we need it:

Similar to what is available for nodeSelector, this adds also the diskSelector for the default storage class created from the chart.

#### Special notes for your reviewer:

First time contributor here, did I miss anything process-wise? Do I need to update the change log, or is that automated?

### CODE REVIEW

**Comment by innobead** [comment]
Thanks for the contribution.

The DCO check failed. Please sign your commit.

**Comment by derekbit** [comment]
@rafaelgaspar Thanks for your contribution.
Can you create an improvement ticket in https://github.com/longhorn/longhorn/issues and add the PR in the ticket?
This will help us maintain and verity the feature.
Thank you.

File: chart/README.md [hunk 1]
```diff
@@ -146,6 +146,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
 | persistence.defaultMkfsParams | string | `""` | mkfs parameters of the default Longhorn StorageClass. |
 | persistence.defaultNodeSelector.enable | bool | `false` | Setting that allows you to enable the node selector for the default Longhorn StorageClass. |
 | persistence.defaultNodeSelector.selector | string | `""` | Node selector for the default Longhorn StorageClass. Longhorn uses only nodes with the specified tags for storing volume data. (Examples: "storage,fast") |
+| persistence.defaultDiskSelector.enable | bool | `false` | Setting that allows you to enable the node selector for the default Longhorn StorageClass. |
```
Comment (jillian-maroket) [inline chart/README.md:149]:
```suggestion
| persistence.defaultDiskSelector.enable | bool | `false` | Setting that allows you to enable the disk selector for the default Longhorn StorageClass. |
```

File: chart/README.md [hunk 2]
```diff
@@ -146,6 +146,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
...
+| persistence.defaultDiskSelector.selector | string | `""` | Disk selector for the default Longhorn StorageClass. Longhorn uses only disk with the specified tags for storing volume data. (Examples: "ssd") |
```
Comment (jillian-maroket) [inline chart/README.md:150]:
```suggestion
| persistence.defaultDiskSelector.selector | string | `""` | Disk selector for the default Longhorn StorageClass. Longhorn uses only disks with the specified tags for storing volume data. (Examples: "ssd,fast") |
```

Comment (jillian-maroket) [reply chart/README.md:150]: The example is from [Disk Selector (field: parameters.diskSelector)](https://longhorn.io/docs/1.6.1/references/storage-class-parameters/#disk-selector-field-parametersdiskselector).

File: chart/questions.yaml [hunk 3]
```diff
@@ -798,6 +798,20 @@ questions:
...
+  description: "Setting that allows you to enable the disk selector for the default Longhorn StorageClass."
+  group: "Longhorn Storage Class Settings"
+  label: Enable Storage Class Disk Selector
+  type: boolean
+  default: false
+  show_subquestion_if: true
+  subquestions:
+  - variable: persistence.defaultDiskSelector.selector
+    label: Storage Class Disk Selector
+    description: 'Disk selector for the default Longhorn StorageClass. Longhorn uses only disks with the specified tags for storing volume data. (Examples: "ssd")'
```
Comment (jillian-maroket) [inline chart/questions.yaml:811]:
```suggestion
    description: "Disk selector for the default Longhorn StorageClass. Longhorn uses only disks with the specified tags for storing volume data. (Examples: "ssd,fast")"
```

File: chart/values.yaml [hunk 4]
```diff
@@ -159,6 +159,11 @@ persistence:
     enable: false
     # -- Node selector for the default Longhorn StorageClass. Longhorn uses only nodes with the specified tags for storing volume data. (Examples: "storage,fast")
     selector: ""
+  defaultDiskSelector:
+    # -- Setting that allows you to enable the disk selector for the default Longhorn StorageClass.
+    enable: false
+    # -- Disk selector for the default Longhorn StorageClass. Longhorn uses only disks with the specified tags for storing volume data. (Examples: "sdd")
```
Comment (jillian-maroket) [inline chart/values.yaml:165]:
```suggestion
    # -- Disk selector for the default Longhorn StorageClass. Longhorn uses only disks with the specified tags for storing volume data. (Examples: "ssd,fast")
```

**Review by jillian-maroket — COMMENTED** [review]
@derekbit Please double-check the details.
@rafaelgaspar Thank you for contributing. Please apply the requested changes.

**Pull request closed without merging**